picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -768,20 +778,18 @@ public Builder listener(@NonNull Listener listener) {
       return this;
     }
 
-    /**
-     * Specify a transformer for all incoming requests.
-     * <p>
-     * <b>NOTE:</b> This is a beta feature. The API is subject to change in a backwards incompatible
-     * way at any time.
-     */
-    public Builder requestTransformer(@NonNull RequestTransformer transformer) {
+    /** Add a transformer that observes and potentially modify all incoming requests. */
+    public Builder addRequestTransformer(RequestTransformer transformer) {
       if (transformer == null) {
         throw new IllegalArgumentException("Transformer must not be null.");
       }
-      if (this.transformer != null) {
+      if (requestTransformers == null) {
+        requestTransformers = new ArrayList<>();
+      }
+      if (requestTransformers.contains(transformer)) {
```

Comment:
do we care about this case, anyway?

Reply:
depends on the transformer impl?

Reply:
i mean how likely is it that somebody accidentally adds the same RequestTransformer twice?

Reply:
bugs happen :)